docs/source/metric_integrations.mdx

```diff
@@ -0,0 +1,136 @@
+
+trainer.train()
+```
+
+### Seq2SeqTrainer
+
+For sequence-to-sequence tasks such as translation or summarization the [`~transformers.Seq2SeqTrainer`] is useful. It allows for the use of the generate method when setting `predict_with_generate=True`. That means we evaluate generated text within the `compute_metric` function, we just need to decode the predictions and labels first.
+
+```python
+import nltk
+from datasets import load_dataset
+import evaluate
+import numpy as np
+from transformers import AutoTokenizer, DataCollatorForSeq2Seq
+from transformers import AutoModelForSeq2SeqLM, Seq2SeqTrainingArguments, Seq2SeqTrainer
+
+# Prepare and tokenize dataset
+billsum = load_dataset("billsum", split="ca_test")
+billsum = billsum.train_test_split(test_size=0.2)
+tokenizer = AutoTokenizer.from_pretrained("t5-small")
+prefix = "summarize: "
+
+def preprocess_function(examples):
+    inputs = [prefix + doc for doc in examples["text"]]
+    model_inputs = tokenizer(inputs, max_length=1024, truncation=True)
+
+    labels = tokenizer(text_target=examples["summary"], max_length=128, truncation=True)
+
+    model_inputs["labels"] = labels["input_ids"]
+    return model_inputs
+
+tokenized_billsum = billsum.map(preprocess_function, batched=True)
+
+# Setup evaluation
+nltk.download("punkt", quiet=True)
+metric = evaluate.load("rouge")
+
+def compute_metrics(eval_preds):
+    preds, labels = eval_preds
+
+    # decode preds and labels
+    labels = np.where(labels != -100, labels, tokenizer.pad_token_id)
+    decoded_preds = tokenizer.batch_decode(preds, skip_special_tokens=True)
+    decoded_labels = tokenizer.batch_decode(labels, skip_special_tokens=True)
+
+    # rougeLSum expects newline after each sentence
+    decoded_preds = ["\n".join(nltk.sent_tokenize(pred.strip())) for pred in decoded_preds]
+    decoded_labels = ["\n".join(nltk.sent_tokenize(label.strip())) for label in decoded_labels]
+
+    result = metric.compute(predictions=decoded_preds, references=decoded_labels, use_stemmer=True)
+    return result
+
+# Load pretrained model and evaluate model after each epoch
+model = AutoModelForSeq2SeqLM.from_pretrained("t5-small")
+data_collator = DataCollatorForSeq2Seq(tokenizer=tokenizer, model=model)
+
+training_args = Seq2SeqTrainingArguments(
+    output_dir="./results",
+    evaluation_strategy="epoch",
+    learning_rate=2e-5,
+    per_device_train_batch_size=16,
+    per_device_eval_batch_size=4,
+    weight_decay=0.01,
+    save_total_limit=3,
+    num_train_epochs=2,
+    fp16=True,
+    predict_with_generate=True
+)
+
+trainer = Seq2SeqTrainer(
+    model=model,
+    args=training_args,
+    train_dataset=tokenized_billsum["train"],
+    eval_dataset=tokenized_billsum["test"],
+    tokenizer=tokenizer,
+    data_collator=data_collator,
+    compute_metrics=compute_metrics
+)
+
+trainer.train()
+```
+
+You can use any `evaluate` metric with the `Trainer` and `Seq2SeqTrainer` as long as they are compatible with the task and predictions.
```

Comment:
Maybe also worth adding an example of running evaluation in an existing model on the Hub

Reply:
I added a comment on this. Essentially replacing `trainer.train` to `trainer.evaluate`.